### PR TITLE
Enable search in the panel

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #1067 

### Summary of Changes

* enables search, which turns on the new search bar in the panel
* update the L10N bootstrap logic so that it expects L10N to be passed in. Previously, the assumption was that L10N was a global on the window, but that was a bad assumption
* there's an associated m-c change as well that i'll link to soon

### CODE REVIEW

File: public/js/main.js [hunk 1]
```diff
@@ -49,11 +49,13 @@ if (isFirefoxPanel()) {
   const prettyPrint = require("./utils/pretty-print");
 
   module.exports = {
-    bootstrap: ({ threadClient, tabTarget, toolbox }) => {
+    bootstrap: ({ threadClient, tabTarget, toolbox, L10N }) => {
```
Comment (jlongster) [inline public/js/main.js]: This is an unrelated change, can you make this a separate PR?

File: public/js/components/Editor.js [hunk 2]
```diff
@@ -266,9 +265,12 @@ const Editor = React.createClass({
       enableCodeFolding: false,
       gutters: ["breakpoints"],
       value: " ",
-      extraKeys
+      extraKeys: {}
     });
 
+    // disables the default search shortcuts
+    this.editor._initShortcuts = () => {};
```
Comment (jlongster) [inline public/js/components/Editor.js]: I'm a little sad that we're diverging on another thing. It would be nice to work on the official editor in m-c instead of customizing it to our needs, but I know that's difficult right now because we need to separate it out into a module for that to really happen.

It's OK for now.

Comment (jasonLaster) [inline public/js/components/Editor.js]: Yeah, my initial change was to the editor.js for that reason, but I didn't want the headache of disabling it there, but not globally for all the tools. This will be better when our Editor Search Bar is global for all the tools `_shrug_`

File: public/js/main.js (hunk 1 above)
Comment (jasonLaster) [inline public/js/main.js]: ugh... just merged this.
